tests/Unit/ExternalVersionUpdate/UpdateTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Unit tests related to external version update.
+ */
+
+namespace WooCommerce\Facebook\Tests\ExternalVersionUpdate;
+
+use WooCommerce\Facebook\Framework\Plugin\Exception as PluginException;
+use WooCommerce\Facebook\Framework\Api\Exception as ApiException;
+use \WP_UnitTestCase;
+
+/**
+ * The External version update unit test class.
+ */
+class UpdateTest extends WP_UnitTestCase {
+
+	/**
+	 * Instance of the Update class that we are testing.
+	 *
+	 * @var \WooCommerce\Facebook\ExternalVersionUpdate\Update The object to be tested.
+	 */
+	private $update;
+
+	/**
+	 * Setup the test object for each test.
+	 */
+	public function setUp():void {
+		$this->update = new \WooCommerce\Facebook\ExternalVersionUpdate\Update();
+	}
+
+	/**
+	 * Test should_update_version
+	 */
+	public function test_should_update_version() {
+		$plugin = facebook_for_woocommerce();
+
+		// Assert update not required when the versions match.
+		update_option( 'facebook_for_woocommerce_latest_version_sent_to_server', \WC_Facebookcommerce_Utils::PLUGIN_VERSION );
```

Comment:
I know that tests commonly call `update_option()`, but I feel that it is a bit cleaner to make use of the `pre_option_{$option}` filter to return the value that we want during the test. This can even be done with an anonymous function, which is then unhooked (rather than resetting the option). Something like this:

```php
$filter = function() {
	return WC_Facebookcommerce_Utils::PLUGIN_VERSION;
};

add_filter( 'pre_option_facebook_for_woocommerce_latest_version_sent_to_server', $filter );

// testing code, then later...
remove_filter( 'pre_option_facebook_for_woocommerce_latest_version_sent_to_server', $filter );

// add it again...
add_filter( 'pre_option_facebook_for_woocommerce_latest_version_sent_to_server', $filter );

// remove it again
remove_filter( 'pre_option_facebook_for_woocommerce_latest_version_sent_to_server', $filter );
```